runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java

```diff
@@ -379,6 +386,16 @@ public void onChatMessage(ChatMessage event)
 				takeScreenshot(fileName, "Duels");
 			}
 		}
+
+		if (chatMessage.equals("You enter the Gauntlet."))
+		{
+			insideGauntlet = true;
+		}
+
+		if (chatMessage.equals("You leave the Gauntlet."))
+		{
+			insideGauntlet = false;
+		}
```

Comment:
These should check for type. Also, is it possible to leave in any way so that the exit message isn't triggered? Logging out, teleporting etc? Maybe it's better to use a region check?